projects/llm_d/runtime/llmd_runtime.py

```
@@ -0,0 +1,695 @@
+    predicate,
+) -> Any:
+    deadline = time.time() + timeout_seconds
+    last_error: Exception | None = None
+
+    while time.time() < deadline:
+        try:
+            value = predicate()
+            if value:
+                return value
+            last_error = None
+        except Exception as exc:  # pragma: no cover - exercised in integration paths
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)